apps/designer/app/designer/features/style-panel/style-source-input/style-source-input.tsx

```diff
@@ -0,0 +1,189 @@
+  Text,
+} from "@webstudio-is/design-system";
+import {
+  ComponentProps,
+  ElementRef,
+  forwardRef,
+  RefObject,
+  useState,
+} from "react";
+import { mergeRefs } from "@react-aria/utils";
+
+export type StyleSource = {
+  id: string;
+  label: string;
+  type: "local" | "token";
+};
+
+type StyleSourceItemProps = {
+  label: string;
+};
+const StyleSourceItem = ({ label }: StyleSourceItemProps) => {
+  return (
+    <Button variant="gray" css={{ maxWidth: "100%" }}>
+      <Text truncate>{label}</Text>
+    </Button>
+  );
+};
+
+type TextFieldWrapperProps = Omit<ComponentProps<"input">, "value"> &
+  Pick<
+    ComponentProps<typeof TextFieldContainer>,
+    "variant" | "state" | "css"
+  > & {
+    value: Array<StyleSource>;
+    inputValue: string;
+    disabled?: boolean;
+    css?: CSS;
+    containerRef?: RefObject<HTMLDivElement>;
+    inputRef?: RefObject<HTMLInputElement>;
+  };
+
+const TextField = forwardRef<ElementRef<typeof Box>, TextFieldWrapperProps>(
+  (props, forwardedRef) => {
+    const {
+      css,
+      disabled,
+      containerRef,
+      inputRef,
+      state,
+      variant,
+      onFocus,
+      onBlur,
+      onClick,
+      type,
+      onKeyDown,
+      inputValue,
+      value,
+      ...textFieldProps
+    } = props;
+    const [internalInputRef, focusProps] = useTextFieldFocus({
+      disabled,
+      onFocus,
+      onBlur,
+    });
+
+    return (
+      <TextFieldContainer
+        {...focusProps}
+        aria-disabled={disabled}
+        ref={mergeRefs(forwardedRef, containerRef ?? null)}
+        state={state}
+        variant={variant}
+        css={{ ...css, px: "$spacing$3", py: "$spacing$2" }}
+        onKeyDown={onKeyDown}
+      >
+        {value.map((item, index) => (
+          <StyleSourceItem label={item.label} key={index} />
+        ))}
+        {/* We want input to be the first element in DOM so it receives the focus first */}
+        <TextFieldInput
+          {...textFieldProps}
+          value={inputValue}
+          type={type}
+          disabled={disabled}
+          onClick={onClick}
+          ref={mergeRefs(internalInputRef, inputRef ?? null)}
+        />
+      </TextFieldContainer>
+    );
+  }
+);
+TextField.displayName = "TextField";
+
+type StyleSourceInputProps = {
+  items?: Array<StyleSource>;
+  value?: Array<StyleSource>;
+  onItemSelect: (item: StyleSource) => void;
+  onItemRemove: (item: StyleSource) => void;
+  onItemCreate: (label: string) => void;
+  css?: CSS;
+};
+
+const initialValue: StyleSource = {
+  label: "",
+  id: "__INITIAL_ID__",
+  type: "local",
+};
+
+export const StyleSourceInput = (props: StyleSourceInputProps) => {
+  const value = props.value ?? [];
+  const [inputValue, setInputValue] = useState(initialValue.label);
+  const {
+    items,
+    getInputProps,
+    getComboboxProps,
+    getMenuProps,
+    getItemProps,
+    isOpen,
+  } = useCombobox({
+    items: props.items ?? [],
+    value: { ...initialValue, label: inputValue },
```

Comment:
Shouldn't value reference match one of "items" item?

Reply:
Made types better, in our case here input value can be intermediate, since user can also create an item, not only select  one existing